delfin/drivers/hpe/hpe_msa/ssh_handler.py

```diff
@@ -479,3 +478,319 @@ def handle_xml_to_dict(xml_info, element):
+                        "storage_id": storage_id,
+                        "native_storage_host_initiator_id":
+                            initiator.get('durable-id'),
+                        "wwn": initiator.get('id'),
+                        "status": constants.InitiatorStatus.ONLINE,
+                        "native_storage_host_id": initiator.get('host-id')
+                    }
+                    initiator_list.append(initiator_item)
+            return initiator_list
+        except Exception as e:
+            LOG.error("Failed to get initiators from msa")
+            raise e
+
+    def list_storage_hosts(self, storage_id):
+        try:
+            hosts_info = self.ssh_pool.do_exec('show host-groups')
+            host_list = []
+            hosts = self.handle_xml_to_json(hosts_info, 'host')
+            host_map = {}
+            for host in hosts:
+                status = constants.HostStatus.NORMAL
+                os_type = constants.HostOSTypes.HP_UX
+                host_member_count = int(host.get('member-count'))
+                if host_member_count > 0:
+                    serial_number = host_map.get(host.get('serial-number'))
+                    if not serial_number:
+                        host_map[host.get('serial-number')] = \
+                            host.get('serial-number')
+                        host_dict = {
+                            "name": host.get('name'),
+                            "description": host.get('durable-id'),
+                            "storage_id": storage_id,
+                            "native_storage_host_id":
+                                host.get('serial-number'),
+                            "os_type": os_type,
+                            "status": status
+                        }
+                        host_list.append(host_dict)
+            return host_list
+        except Exception as e:
+            LOG.error("Failed to get host from msa")
+            raise e
+
+    def list_storage_host_groups(self, storage_id):
+        try:
+            host_groups_info = self.ssh_pool.do_exec('show host-groups')
+            storage_host_groups_result = {}
+            host_group_list = []
+            storage_host_grp_relation_list = []
+            host_groups = self.handle_xml_to_json(
+                host_groups_info, 'host-group')
+            host_info_list = self.handle_xml_to_json(host_groups_info, 'host')
+            if host_groups:
+                for host_group in host_groups:
+                    hosts_list = []
+                    storage_host_group_id = host_group.get('serial-number')
+                    if host_info_list:
+                        for host_info in host_info_list:
+                            host_id = host_info.get('serial-number')
+                            host_group_id = host_info.get('host-group')
+                            if host_id != 'NOHOST':
+                                if host_group_id == storage_host_group_id:
```

Comment:
I think these two `if` should be written like `if host_id != 'NOHOST' and host_group_id == storage_host_group_id` .

Reply:
These two judgments are combined
![image](https://user-images.githubusercontent.com/88076879/155675906-008a4761-f3d7-46b6-b0df-6f15decb98b7.png)